File: 6-kyu/ip-address-to-number/javascript/solution.js
const ipToNum = ip => {
  const octets = ip.split(".").map(n => Number(n).toString(2).padStart(8, '0'));
  const joined = octets.join("");
  const baseTen = Number.parseInt(joined, 2);
  return baseTen;
};

const numToIp = num => {
  const baseTwo = num.toString(2).padStart(32, '0');
  const octets = [];
  for (let i = 0; i < baseTwo.length; i += 8) {
    const octet = baseTwo.slice(i, i + 8);
    octets.push(octet);
  }
  return octets.map(n => Number.parseInt(n, 2)).join(".");
};

File: 6-kyu/ip-address-to-number/python/solution.py
def ip_to_num(ip):
    octets = [f'{int(octet):08b}' for octet in ip.split('.')]
    joined = ''.join(octets)
    return int(joined, 2)

def num_to_ip(num):
    num_bin = f'{num:b}'.zfill(32)
    octets = [f'{int(num_bin[idx:idx + 8], 2)}' for idx in range(0, len(num_bin), 8)]
    return '.'.join(octets)
